### DISCUSSION TITLE
friendly urls

### DISCUSSION THREAD

**Discussion by rogerg1967** [openstore-ecommerce/OpenStore#243] [Q&A]
Hi,
I want to rewrite the URLs for categories and products, for example:
/amusement-games-for-hire/catid/79/retro-games-for-hire
becomes
/retro-games-for-hire
i have been asked to do this by an SEO company. Please advise the best way to do this?
Regards
Roger

**Comment by leedavi** [comment]
Aren't SEO companies fun.  Giving you worthless work for nothing. :-)

I presume "/amusement-games-for-hire" is a page with the product module on it?

The problem is that the module needs the category id in the URL to make the selection.  You could re-write the module to work on a defined list, but I think the quickest way would be to create a page call "/retro-games-for-hire", put the product list module on it with a default of "catid" 79 and a page detail in the settings.

**Comment by rogerg1967** [comment]
Thats great thank you, I will create pages for each category.

**Comment by DigitalTrading** [comment]
Hi Dave,
We have written a url extension provider which now generates friendly urls. You can see them in action at:
https://www.printableparties.com
Let me know if you think this is a reasonable solution or if there could be any issues.
Thanks

**Reply by leedavi** [reply]
It look s great, the URLs work and I don't think there will be an issue.

Although I'm unsure how exactly you have done it.  With a managed Handles modules in the web.config ?

Are you willing to share your solution?

**Comment by DigitalTrading** [comment]
Hi Dave,
Thanks for taking a look and yes I will share this with you. All being well once we are happy it is working effectively we can add this as a an optional URL provider.
Will update you soon.
Regards
Roger

**Reply by leedavi** [reply]
OK Great.  I sent an email to your hello@ email address.  Get in touch with me from that.
